File: apps/mobile_flutter/lib/features/language_selection/domain/language_repository.dart
import '../../../core/api/api_client.dart';
import 'language_pack.dart';

class LanguageRepository {
  const LanguageRepository(this.apiClient);

  final ApiClient apiClient;

  Future<List<LanguagePack>> fetchLanguages() async {
    final rows = await apiClient.getList('/languages');
    return rows
        .cast<Map<String, dynamic>>()
        .map(LanguagePack.fromJson)
        .toList(growable: false);
  }
}

File: apps/web_pwa/src/features/languages/languageRepository.ts
import type { ApiClient } from '../../core/api/apiClient';
import { languagePackFromJson, type LanguagePack } from './types';

// Mirrors apps/mobile_flutter/lib/features/language_selection/domain/language_repository.dart.
export class LanguageRepository {
  private readonly apiClient: ApiClient;

  constructor(apiClient: ApiClient) {
    this.apiClient = apiClient;
  }

  async fetchLanguages(): Promise<LanguagePack[]> {
    const rows = await this.apiClient.getList<Record<string, unknown>>('/languages');
    return rows.map(languagePackFromJson);
  }
}
